.github/workflows/wheels.yml

```diff
@@ -142,6 +161,7 @@ jobs:
                             OPENBLAS_NUM_THREADS=2
                             SKLEARN_SKIP_NETWORK_TESTS=1
                             SKLEARN_BUILD_PARALLEL=3
+                            PIP_EXTRA_INDEX_URL=https://pypi.anaconda.org/scipy-wheels-nightly/simple
```

Comment:
As described in https://github.com/scikit-learn/scikit-learn/pull/24446#issuecomment-1273357186, including the nightly index is good for testing, but not mergable.

Now that Python 3.11 wheels for [SciPy](https://pypi.org/project/scipy/#files) and [NumPy](https://pypi.org/project/numpy/#files) are on the public PyPI, we should be able to download them from there.

Reply:
It will allow to test the wheel build with nightly build, otherwise this is not tested elsewhere (scipy_dev nightly CI only tests the build and test of scikit-learn without wheels): for instance the lack of `msvcp140.dll` was already detected by this PR before the nightly build error #24612.

Reply:
> https://github.com/scikit-learn/scikit-learn/issues/24612 is another (new) unrelated problem that we also need to fix. I do not yet understand what's going on but let's discuss it on the https://github.com/scikit-learn/scikit-learn/issues/24612 issue to avoid crossing discussions.

From https://github.com/scikit-learn/scikit-learn/pull/24446#issuecomment-1273357186: the two issues were in fact exactly the same problem already raised here and not at all unrelated. The question is not "is this mergeable or not" (what does that mean anyway?) but "when do you want to catch the issue": as soon as possible or after the publication of the wheels from other projects.

Reply:
There are two related concerns here:

1. Run CI for wheels built with nightly dependencies
2. Publish wheels that are tested on dependencies that were released on PyPI (not the nightly index)

If we enable downloading from the nightly index all the time we resolve 1. But then we may publish wheels that were tested on the nightly dependencies and not the ones released on PyPI.

In our case, I think we can make `PIP_EXTRA_INDEX_URL` conditioned on it being a nightly build. Specifically:

1. If nightly build, use `PIP_EXTRA_INDEX_URL=https://pypi.anaconda.org/scipy-wheels-nightly/simple`
2. If not nightly build, do not set `PIP_EXTRA_INDEX_URL`.